internal/services/apimanagement/api_management_resource.go

```diff
@@ -741,6 +745,66 @@ func resourceApiManagementServiceCreateUpdate(d *pluginsdk.ResourceData, meta in
 
 	d.SetId(id.ID())
 
+	// Remove sample products and APIs after creating (v3.0 behaviour)
```

Comment:
one final thought, should we be adding a blue info box to the docs about this? something along the lines of:

```
-> When creating a new API Management resource in version 3.0 of the AzureRM Provider and later, please be aware that the AzureRM Provider will now cleanup any sample APIs and Products created by the Azure API during the creation of the API Management resource.
```

Reply:
Yeah definitely, was on the fence whether to mention this only in the upgrade guide but agree it's worth calling out in the resource doc